### PR TITLE
Change version to be dynamically synced

### PR DESCRIPTION
**Pull request by drazisil**
Currently the version of the package is not in sync and is reporting 2.0.21, instead of 2.1.2

This PR corrects that by moving the version to a central location.

### CODE REVIEW

**Comment by drazisil** [comment]
Note, the version being 2.1.3 is not a typo, that's the version that will be cut when this is merged.

**Comment by drazisil** [comment]
Coverage is also incorrect as it was reporting 0% base coverage on master from broken version 2.0.21. The coverage amount should not have changed as no tests or testable code were modified in this PR.

File: .gitignore [hunk 1]
```diff
@@ -12,3 +12,5 @@ dist
 htmlcov
 venv
 *.sw[op]
+
+.idea/
```
Comment (thomasrockhu) [inline .gitignore:16]: does this have to be here?

File: .idea/.gitignore [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# Default ignored files
```
Comment (thomasrockhu) [inline .idea/.gitignore:1]: same with this file

File: codecov/__init__.py [hunk 3]
```diff
@@ -9,6 +9,10 @@
 from time import sleep
 from json import loads
 
+from .__version__ import __title__, __description__, __url__, __version__
+from .__version__ import __author__, __author_email__, __license__
+from .__version__ import __copyright__
```
Comment (thomasrockhu) [inline codecov/__init__.py:14]: prefer

```suggestion
from .__version__ import (
    __author__,
    __author_email__,
    __copyright__,
    __description__,
    __license__,
    __title__,
    __url__,
    __version__,
)
```

don't LOVE that we do an import from `.__version__`

File: codecov/__init__.py [hunk 4]
```diff
@@ -988,7 +992,7 @@ def main(*argv, **kwargs):
             )
         )
 
-        query["package"] = "py" + VERSION
+        query["package"] = "py" + __version__
```
Comment (thomasrockhu) [inline codecov/__init__.py:995]:
```suggestion
        query["package"] = "py" + version
```

File: codecov/__version__.py [hunk 5]
```diff
@@ -0,0 +1,8 @@
+__title__ = 'codecov'
```
Comment (thomasrockhu) [inline codecov/__version__.py:1]: nit: alpha this please

File: setup.py [hunk 6]
```diff
@@ -1,7 +1,10 @@
 #!/usr/bin/env python
 from setuptools import setup
 
-version = "2.1.1"
+from codecs import open
```
Comment (thomasrockhu) [inline setup.py:4]:
```
import os

from codecs import open
from setuptools import setup
```

File: setup.py [hunk 7]
```diff
@@ -19,17 +22,23 @@
     "Topic :: Software Development :: Testing",
 ]
 
+here = os.path.abspath(os.path.dirname(__file__))
```
Comment (thomasrockhu) [inline setup.py:25]: :s/here/filepath/g

File: codecov/__init__.py (hunk 3 above)
Comment (drazisil) [reply codecov/__init__.py:14]: Why? That's the whole goal, so we have a single place for both pip and the package.

File: codecov/__version__.py (hunk 5 above)
Comment (drazisil) [reply codecov/__version__.py:1]: I'm not sure we want to change the package name, won't that break everyone's import?

File: codecov/__init__.py (hunk 3 above)
Comment (thomasrockhu) [reply codecov/__init__.py]: I take that back. Absolute imports are preferred, but PEP-8 says 
```However, explicit relative imports are an acceptable alternative to absolute imports, especially when dealing with complex package layouts where using absolute imports would be unnecessarily verbose:```

File: .gitignore (hunk 1 above)
Comment (drazisil) [reply .gitignore:16]: Unless you want commiting my .idea dir contents

Comment (thomasrockhu) [reply .gitignore:16]: doesn't seem like it should exist in the repo though

File: .idea/.gitignore (hunk 2 above)
Comment (drazisil) [reply .idea/.gitignore]: That should not be there, it got missed.

File: codecov/__version__.py (hunk 5 above)
Comment (thomasrockhu) [reply codecov/__version__.py:1]: errrr, I mean
```
  1 __author__ = 'Codecov'
  2 __author_email__ = 'support@codecov.io'
  3 __copyright__ = 'Copyright 2020 Codecov'
  4 __description__ = 'Hosted coverage reports for GitHub, Bitbucket and Gitlab'
  5 __license__ = 'Apache 2.0'
  6 __title__ = 'codecov'
  7 __url__ = 'https://github.com/codecov/codecov-python'
  8 __version__ = '2.1.3'
```

File: .gitignore (hunk 1 above)
Comment (drazisil) [reply .gitignore:16]: IntelliJ always puts it's .idea directory in the repos.

File: setup.py [hunk 8]
```diff
@@ -1,7 +1,8 @@
 #!/usr/bin/env python
+from codecs import open
+import os
 from setuptools import setup
```
Comment (thomasrockhu) [inline setup.py:4]: super nit
```suggestion
import os

from setuptools import setup
```

**Pull request merged**